File: src/main/java/com/ject/studytrip/global/common/response/StandardResponse.java
package com.ject.studytrip.global.common.response;

import com.ject.studytrip.global.exception.response.ErrorResponse;

public record StandardResponse(boolean success, int status, Object data) {

    public static StandardResponse success(int status, Object data) {
        return new StandardResponse(true, status, data);
    }

    public static StandardResponse fail(int status, ErrorResponse errorResponse) {
        return new StandardResponse(false, status, errorResponse);
    }
}

File: src/main/kotlin/com/ject/studytrip/global/common/response/StandardResponse.kt
package com.ject.studytrip.global.common.response

import com.ject.studytrip.global.exception.response.ErrorResponse

data class StandardResponse(
    val success: Boolean,
    val status: Int,
    val data: Any?,
) {
    companion object {
        fun success(
            status: Int,
            data: Any?,
        ): StandardResponse = StandardResponse(true, status, data)

        fun fail(
            status: Int,
            errorResponse: ErrorResponse,
        ): StandardResponse = StandardResponse(false, status, errorResponse)
    }
}
